File: 0258-add-digits/0258-add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        
        if(num<=9)return num;
        int givenNumber=num;
        int sum = 0;
        
         while(1){
            while(givenNumber>0)
            {
                int lastDigit = givenNumber%10;
                
                sum += lastDigit;
                
                givenNumber/=10;
            }
             
           if (sum<=9)return sum;
           givenNumber = sum;
           sum = 0;
         }
        return -1;
    
    }
};

File: 0258-add-digits/0258-add-digits.go
func addDigits(num int) int {
    
    if num<=9{
            return num
        }
    
    sum := 0
    for{
        for num>0{
            lastDigit := num%10
            sum += lastDigit
            num /= 10
        }
        if sum<=9{
            return sum
        }
        num = sum
        sum = 0
    }
}